pkg/resource/unstructured/composite/composite.go

```diff
@@ -218,3 +218,25 @@ func (c *Unstructured) GetConnectionDetailsLastPublishedTime() *metav1.Time {
 func (c *Unstructured) SetConnectionDetailsLastPublishedTime(t *metav1.Time) {
 	_ = fieldpath.Pave(c.Object).SetValue("status.connectionDetails.lastPublishedTime", t)
 }
+
+// GetEnvironmentConfigReferences of this Composite resource.
+func (c *Unstructured) GetEnvironmentConfigReferences() []corev1.ObjectReference {
+	out := &[]corev1.ObjectReference{}
+	_ = fieldpath.Pave(c.Object).GetValueInto("spec.environmentConfigRefs", out)
+	return *out
+}
+
+// SetEnvironmentConfigReferences of this Composite resource.
+func (c *Unstructured) SetEnvironmentConfigReferences(refs []corev1.ObjectReference) {
+	empty := corev1.ObjectReference{}
+	filtered := make([]corev1.ObjectReference, 0, len(refs))
+	for _, ref := range refs {
+		// TODO(negz): Ask muvaf to explain what this is working around. :)
+		// TODO(muvaf): temporary workaround.
```

Comment:
I think @muvaf is on vacation, but I'd still love to know what this is doing. 😄

Reply:
I think it was about having empty elements in the array causing problem since the element would be omitted but I'm not 100% sure.